### PR TITLE
Create signinfo refactor proper

### PR DESCRIPTION
**Pull request by mandeepsandhu**
# Issue

What issue is this PR targeting? If there is no issue that addresses the problem, please open a corresponding issue and link it here.

## Tasklist

 - [ ] Add tests
 - [ ] Add #fixes with the issue number that this PR addresses
 - [ ] Update the docs with any new request parameters or changes to behavior described
 - [ ] Update the [changelog](CHANGELOG.md)
 - [ ] If you made changes to the lua files, update the [taginfo](taginfo.json) too.

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: test/gurka/test_phonemes.cc [hunk 1]
```diff
@@ -377,25 +377,24 @@ TEST(Standalone, PhonemesWithAltandDirection) {
     std::vector<SignInfo> signs = tile->GetSigns(node_id.id(), pronunciations, true);
 
     uint32_t sign_index = 0;
-    EXPECT_NE(signs.size(), 0);
+    ASSERT_EQ(signs.size(), 1);
```
Comment (mandeepsandhu) [inline test/gurka/test_phonemes.cc:380]: The tests were modified by running it against pre-refactored code and capturing expected values. Then the same test with stricter checks was run against refactored code to ensure all tests passed.

File: test/gurka/test_phonemes.cc [hunk 2]
```diff
@@ -450,6 +449,7 @@ TEST(Standalone, PhonemesWithAltandDirection) {
           name_pronunciations.find(name_index);
       if (iter == name_pronunciations.end()) {
         if (name_index != 0)
+          // Whats the point of this? this will always be true
```
Comment (mandeepsandhu) [inline test/gurka/test_phonemes.cc:452]: @gknisely there are many places in this test where we are doing such checks. The following `EXPECT_EQ` never fails. Can we check for specific `name_index` instead for which we expect the pronunciation to **not** exist?

I've left comments in other places below for similar conditions. Let me know what you think?

**Review by gknisely — APPROVED** [review]

**Comment by kevinkreiser** [comment]
@mandeepsandhu do you mind putting at least a small description in the pr so those who arent in the know can tell what the purpose was here?

**Comment by mandeepsandhu** [comment]
> @mandeepsandhu do you mind putting at least a small description in the pr so those who arent in the know can tell what the purpose was here?

Hey @kevinkreiser , I updated the PR description. Let me know if you need more added. Essentially the refactor removes a lot of duplicated code and puts it in one place. Also updated tests to make them more stricter in what they test. HTH.

**Comment by kevinkreiser** [comment]
@mandeepsandhu nope thats all i wanted. i kind of t hought it was all just code motion, but its always nice to have a description just to be able to know for sure. thank you!

**Comment by mandeepsandhu** [comment]
> its always nice to have a description just to be able to know for sure. thank you!

:100:

**Pull request merged**